pkg/featuremanager/feature_manager.go

```diff
@@ -0,0 +1,157 @@
+	Feature Feature
+	Enabled bool
+}
+
+type FeatureManager interface {
+	Append(feature Feature, hook cell.HookInterface)
+	List() []Feature
+
+	start(feature Feature, ctx context.Context) error
+	stop(feature Feature, ctx context.Context) error
+	process(status []FeatureStatus, ctx context.Context) error
+}
+
+type DefaultFeatureManager struct {
+	featureLifecycle FeatureLifecycle
+	logger           *slog.Logger
+}
+
+func (d *DefaultFeatureManager) List() []Feature {
+	return d.featureLifecycle.List()
+}
+
+// Append adds the specified hook to the list of hooks managed for the given feature
+// The lifecycle will be managed by the FeatureManager
+func (d *DefaultFeatureManager) Append(feature Feature, hook cell.HookInterface) {
+	if err := d.featureLifecycle.Append(feature, hook); err != nil {
+		d.logger.Error("Failed to append feature", "feature", feature, logfields.Error, err)
+		return
+	}
+}
+
+func (d *DefaultFeatureManager) start(feature Feature, ctx context.Context) error {
+	if err := d.featureLifecycle.Start(feature, ctx, d.logger); err != nil {
+		d.logger.Error("Failed to start feature", "feature", feature, logfields.Error, err)
+		return err
+	}
+	return nil
+}
+
+func (d *DefaultFeatureManager) stop(feature Feature, ctx context.Context) error {
+	if err := d.featureLifecycle.Stop(feature, ctx, d.logger); err != nil {
+		d.logger.Error("Failed to stop feature", "feature", feature, logfields.Error, err)
+		return err
+	}
+	return nil
+}
+
+func (d *DefaultFeatureManager) process(status []FeatureStatus, ctx context.Context) error {
+	// The implementation leaks some safety features when enabling compute intensive features
+	// see https://github.com/cilium/cilium/issues/34930
+	d.logger.Info("Processing feature status", "features", status)
+
+	var errs error
+	for _, feature := range status {
+		var err error
+		if feature.Enabled {
+			err = d.start(feature.Feature, ctx)
+		} else {
+			err = d.stop(feature.Feature, ctx)
+		}
+
+		if err != nil {
+			errs = errors.Join(errs, err)
+		}
+	}
+
+	return errs
+}
+
+type params struct {
+	cell.In
+
+	JobGroup                job.Group
+	FeatureManager          FeatureManager
+	Logger                  *slog.Logger
+	Lifecycle               cell.Lifecycle
+	Table                   statedb.Table[dynamicconfig.DynamicConfig]
+	DB                      *statedb.DB
+	Config                  config
+	DynamicConfigCellConfig dynamicconfig.Config
+}
+
+func provideFeatureManager(log *slog.Logger) FeatureManager {
+	return &DefaultFeatureManager{
+		featureLifecycle: NewFeatureLifecycle(),
+		logger:           log,
+	}
+}
+
+func registerFeatureManager(p params) {
+	if !p.Config.EnableFeatureManager || !p.DynamicConfigCellConfig.EnableDynamicConfig {
+		return
+	}
+
+	p.JobGroup.Add(job.OneShot("feature-manager-table-watcher", func(ctx context.Context, health cell.Health) error {
+		for {
+			entry, found, w := dynamicconfig.WatchKey(p.DB.ReadTxn(), p.Table, FeatureStatusConfigKey)
```

Comment:
Hmm, this might be nicer if we pull this out into its own thing and break the direct dependency to `dynamicconfig`. E.g. depend on interface that gives the enabled features and a watch channel, and then implement that bridge between `dynamicconfig` and this in `cell.go` or even in `daemon/cmd/cells.go`.